src/vehiclelist.cpp

```diff
@@ -116,8 +116,11 @@ bool GenerateVehicleSortList(VehicleList *list, const VehicleListIdentifier &vli
 
 	switch (vli.type) {
 		case VL_STATION_LIST:
-			for (const Vehicle *v : Vehicle::Iterate()) {
-				if (v->type == vli.vtype && v->IsPrimaryVehicle()) {
+			for (const Company *c : Company::Iterate()) {
+				/* Only iterate over all companies in the case of neutral Oil Rig stations */
+				if (vli.vtype != VEH_AIRCRAFT && vli.vtype != VEH_SHIP && vli.company != c->index) continue;
```

Comment:
I'm not sure if this is correct. Shouldn't it rather test for the station owner being the neutral company, and not care about the vehicle type in question at all?

I also somewhat doubt this is going to be a performance improvement in this specific case. On the other hand, it's "only" a GUI use and not performance critical for the simulation itself.

Reply:
Fixed. Thanks.

There's still a positive gain. In the worst case, it iterates only one vehicle type per company.